src/NATS.Client/Conn.cs

```diff
@@ -987,6 +988,27 @@ public string ConnectedUrl
             }
         }
 
+        /// <summary>
+        /// Gets the IP of client as known by the NATS server, otherwise <c>null</c>.
+        /// </summary>
+        /// <remarks>
+        /// Supported in the NATS server version 2.1.6 and above.  If the client is connected to
+        /// an older server or is in the process of connecting, null will be returned.
+        /// </remarks>
+        public IPAddress ClientIP
+        {
+            get
+            {
+                lock (mu)
+                {
+                    if (string.IsNullOrEmpty(info.client_ip))
+                        return null;
+
+                    return IPAddress.Parse(info.client_ip);
+                }
```

Comment:
We can avoid locking during the IP address parsing:
```suggestion
                string clientIp;
                lock (mu)
                {
                    clientIp = info.client_ip;
                }
                
                return !String.IsNullOrEmpty(clientIp) ? IPAddress.Parse(clientIp) : null;
```

Reply:
Thanks - good catch.